src/main/java/billgates/usecases/user_join/UserJoinUseCase.java

```diff
@@ -0,0 +1,27 @@
+package billgates.usecases.user_join;
+
+import billgates.database.QueryUserData;
+import billgates.interface_adapters.DatabaseGateway;
+
+public class UserJoinUseCase implements UserJoinInputPort{
+    private final DatabaseGateway gateway;
+    private final UserJoinOutputPort outputPort;
+
+    public UserJoinUseCase(DatabaseGateway gateway, UserJoinOutputPort outputPort){
+        this.gateway = gateway;
+        this.outputPort = outputPort;
+    }
+
+    @Override
+    public void join(UserJoinRequestModel model) {
+        QueryUserData userData = gateway.getUserData();
+        if (!userData.getUsers().contains(model.getUsername())){
+            outputPort.display(new UserJoinResponseModel(false, "Not in database"));
+
+        }
+        else {
+            join(model);
```

Comment:
I don't really understand why you call this function recursively.

If the user is in the database, you should pop a dialog and notify the user that the username is invalid.